contracts/mechs/nevermined/BalanceTrackerNvmSubscription.sol

```diff
@@ -113,66 +85,64 @@ contract BalanceTrackerNvmSubscription {
             revert InsufficientBalance(subscriptionBalance, balance);
         }
 
-        // Adjust requester balance
-        mapRequesterBalances[requester] = balance;
+        return balance;
     }
 
-    /// @dev Finalizes mech delivery rate based on requested and actual ones.
-    /// @param mech Delivery mech address.
+    /// @dev Adjusts final requester balance accounting for possible delivery rate difference (credit).
     /// @param requester Requester address.
-    /// @param requestId Request Id.
-    /// @param maxDeliveryRate Requested max delivery rate.
-    function finalizeDeliveryRate(address mech, address requester, uint256 requestId, uint256 maxDeliveryRate) external {
-        // Reentrancy guard
-        if (_locked > 1) {
-            revert ReentrancyGuard();
-        }
-        _locked = 2;
-
-        // Check for marketplace access
-        if (msg.sender != mechMarketplace) {
-            revert MarketplaceOnly(msg.sender, mechMarketplace);
-        }
-
-        // Get actual delivery rate
-        uint256 actualDeliveryRate = IMech(mech).getFinalizedDeliveryRate(requestId);
+    /// @param rateDiff Delivery rate difference.
+    /// @return Adjusted balance.
+    function _adjustFinalBalance(address requester, uint256 rateDiff) internal virtual override returns (uint256) {
+        uint256 balance = mapRequesterBalances[requester];
 
-        // Check for zero value
-        if (actualDeliveryRate == 0) {
-            revert ZeroValue();
+        // This must never happen as max delivery rate is always bigger or equal to the actual delivery rate
+        if (rateDiff > balance) {
+            revert Overflow(rateDiff, balance);
         }
 
-        uint256 rateDiff;
-        if (maxDeliveryRate > actualDeliveryRate) {
-            // Return back requester overpayment credit
-            rateDiff = maxDeliveryRate - actualDeliveryRate;
+        // Adjust requester credit balance
+        return (balance - rateDiff);
+    }
 
-            // Get requester balance
-            uint256 balance = mapRequesterBalances[requester];
+    // TODO: behavior with buyBackBurner?
```

Comment:
what's the buy back burner got to do with this level of abstraction?

Reply:
This is the question - what do we do with collected NVM subscription credits as fees? Need to implement that scenario.

Reply:
This is resolved, commented in following PRs somewhere. NVM subscription is ultimately paid to mechs and MM fee in native / ERC20 assets. This will inherit the behavior of fixed price mechs.